state.go

```diff
@@ -894,26 +908,40 @@ func (m *Memberlist) suspectNode(s *suspect) {
 	changeTime := time.Now()
 	state.StateChange = changeTime
 
-	// Setup a timeout for this
-	timeout := suspicionTimeout(m.config.SuspicionMult, m.estNumNodes(), m.config.ProbeInterval)
-	time.AfterFunc(timeout, func() {
+	// Setup a suspicion timer. Given that we don't have any known phase
+	// relationship with our peers, we set up k such that we hit the nominal
+	// timeout two probe intervals short of what we expect given the suspicion
+	// multiplier. If there aren't enough peers then we just use the nominal
+	// timer, since we know this algorithm no longer makes sense.
+	k := m.config.SuspicionMult - 2
```

Comment:
I feel like we should move some of this logic into a helper function as its pretty tricky